packages/react-devtools-extensions/src/parseHookNames/parseHookNames.js

```diff
@@ -104,12 +128,73 @@ const originalURLToMetadataCache: LRUCache<
   },
 });
 
+function withSyncProfiling<TReturn>(
```

Comment:
This function and the `withCallbackProfiling` function below could be part of such a logger. (Maybe non-Facebook builds would just get a no-op logger that does nothing.)